AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -196,6 +199,17 @@ class SyncMutationToCloudOperation: Operation {
             return
         }
 
+        // sync version in the mutation event table before calling finish()
+        if case let .success(graphQLResponse) = cloudResult {
+            if case let .success(graphQLResult) = graphQLResponse {
+                MutationEvent.updatePendingMutationEventVersion(for: mutationEvent.modelId,
+                                                                mutationSync: graphQLResult,
+                                                                storageAdapter: storageAdapter) { result in
+                    self.log.verbose("\(#function) received result after syncing version from API: \(result)")
+                }
+            }
+        }
+
         finish(result: cloudResult)
```

Comment:
like 
```
MutationEvent.updatePendingMutationEventVersion(for: mutationEvent.modelId,
                                                            mutationSync: graphQLResult,
                                                            storageAdapter: storageAdapter) { result in
                self.log.verbose("\(#function) received result after syncing version from API: \(result)")
                finish(result: cloudResult)
            }
```

otherwise, the operation will finish executing while the MutationEvent.updatePendingMutationEventVersion is being performed asynchronously